### PR TITLE
[JENKINS-73563] create a jenkins-button instead of a yui button in makeButton

### PR DESCRIPTION
**Pull request by mawinter69**
See JENKINS-73563

makeButton is one of the last places in core were YUI is used. This method is called by a few plugins directly but also when plugins create inputs of type button or submit with class `submit-button` or `yui-button`. Frequently they then also attach an onclick handler.

This change will replace the input with a button with class `jenkins-button` on the fly.
Looking at the code of several plugins that rely on the behaviour it seems they will not be affected by this change (tested with shelve-plugin)

credentials plugin works with the returned YUI object and try to set the button to disabled or get the associated form, for this case a small wrapper is added that implements the 2 methods and credentials plugin still works without problems.
There is PR open for credentials that cleans up the code and removes yui usage
(Another usage affects the multi-slave-config plugin which is deprecated and not usable anyway due to prototype usage)

See https://docs.google.com/spreadsheets/d/1UjvtFmNmEdjMN5DUoFxJfBryA8q-E5_HwOzVKbVG9b0/edit?usp=sharing for a complete list of plugins that use makeButton directly or indirectly.

Need to test if ath needs to be adjusted

### Testing done
Manual testing

### Proposed changelog entries

- makeButton creates jenkins-buttons on the fly instead of using YUI

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (timja) [inline war/src/main/webapp/scripts/hudson-behavior.js:915]: docs above could do with an update as it no longer is a YUI button

**Review by timja — COMMENTED** [review]
code lgtm, will approve once i've had time to manually test it

**Comment by mawinter69** [comment]
ath is passing

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (Kevin-CB) [inline war/src/main/webapp/scripts/hudson-behavior.js:948]:
```suggestion
  button.innerText = e.value;
```
Any specific reason to use `innerHTML`? AFAIU, it's just to render the String inside the button.
A simple `innerText` should be sufficient and reduce the potential attack vector for XSS.

**Review by Kevin-CB — APPROVED** [review]
Thanks for the change, LGTM security wise

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by basil** [comment]
Breaks PCT:

- `hudson.security.LDAPEmbeddedTest#validateUI`
- `org.jenkinsci.plugins.docker.commons.credentials.DockerServerDomainSpecificationTest#configRoundTrip`

I am reverting this commit so that the 2.472 weekly release can be delivered tomorrow with Jetty 12 (EE 8) and without PCT failures. This can be reintegrated when there are no PCT failures.

**Comment by MarkEWaite** [comment]
I've temporarily added the `skip-changelog` label so that this commit won't be included in the Jenkins 2.472 changelog.  When it is reintegrated, we'll include it in the changelog for the matching release.

**Comment by mawinter69** [comment]
Seems that htmlunit doesn't support the `forEach` method of a `DOMTokenList`. Should be no big deal to rewrite the code to avoid the forEach and instead loop over the list manually

**Pull request merged**
